fluentd/vendored_gem_src/fluent-plugin-prometheus/lib/fluent/plugin/in_prometheus_tail_monitor.rb

```diff
@@ -38,24 +38,31 @@ def configure(conf)
         @base_labels[key] = expander.expand(value)
       end
 
-      if defined?(Fluent::Plugin) && defined?(Fluent::Plugin::MonitorAgentInput)
-        # from v0.14.6
-        @monitor_agent = Fluent::Plugin::MonitorAgentInput.new
-      else
-        @monitor_agent = Fluent::MonitorAgentInput.new
-      end
+      @monitor_agent = Fluent::Plugin::MonitorAgentInput.new
     end
 
     def start
       super
 
       @metrics = {
-        position: @registry.gauge(
+        position: get_gauge(
           :fluentd_tail_file_position,
           'Current position of file.'),
-        inode: @registry.gauge(
+        inode: get_gauge(
           :fluentd_tail_file_inode,
           'Current inode of file.'),
+        maxfsize: get_gauge(
+          :maxfsize,
+          'Current max fsize of file on rotation event'),
+        countonrotate: get_gauge(
+          :countonrotate,
+          'No of rotation noticed by fluentd'),
+        totalbytesread: get_gauge(
+          :totalbytesread,
+          'totalbytes read by fluentd IOHandler'),
+        totalbytesavailable: get_gauge(
+          :totalbytesavailable,
+          'totalbytes available at each instance of rotation - to be read by fluentd IOHandler'),
```

Comment:
I am not sure that I understand the term ... maybe something with the words

Reply:
I changed the terms in the JIRA, @eranra  if you agree these are more expressive we should update the code:

    bytes_logged (counter) bytes written to log files by containers.
    bytes_collected (counter) bytes read by the collector for forwarding. 

Log loss can be computed by a prometheus query expression: (bytes_logged - bytes_collected)

@pmoogi-redhat note the metrics should be _counters_ not _guages_. They always increase.

Reply:
@alanconway yes ... those terms are simple and make sense :-)

Reply:
changed parameter names from totalbytesavailable and totalbytesread to  totalbytes_logged_in_disk
 &  totalbytes_collected_from_disk respectively. Got a separate class implementation to monitor stat for getting totalbytes_logged_in_disk